tools/functional-tester/etcd-tester/cluster.go

```diff
@@ -266,12 +266,23 @@ func setHealthKey(us []string) error {
 		if err != nil {
 			return fmt.Errorf("%v (%s)", err, u)
 		}
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 		kvc := pb.NewKVClient(conn)
-		_, err = kvc.Put(ctx, &pb.PutRequest{Key: []byte("health"), Value: []byte("good")})
-		cancel()
+		for i := 0; i < 5; i++ {
```

Comment:
there are a lot of code duplication that can be reduced easily. can you make the retry a func?